### PR TITLE
feat: Material UI - Styling, Dialog, and Form Validation(#40835)

### PR DESCRIPTION
**Pull request by nxfareed**
Link to the task: https://redmine.successive.tech/issues/40835

### CODE REVIEW

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import * as yup from 'yup';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import PersonIcon from '@material-ui/icons/Person';
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: use a single import statement for all this.

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+import PropTypes from 'prop-types';
+import EmailIcon from '@material-ui/icons/Email';
+// import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: use a single import statement for all this

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        if (err.message !== error[field]) {
+          this.setState({
+            error: {
+              ...error,
+              [field]: err.message,
+            },
+          });
+        }
+      });
+    }
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: use try-catch and async-await instead

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  render() {
+    const { classes } = this.props;
+    const { open, onClose, onSubmit } = this.props;
+    const {
+      name, email, password, confirmPassword, hasError, error,
+    } = this.state;
+    console.log(this.state);
+    this.hasErrors();
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: Why was this called hasError()

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                variant="outlined"
+              />
+            </Grid>
+            <Grid item xs={6}>
+              <TextField
+                label="Password"
+                id="outlined-start-adornment"
+                type="password"
+                value={password}
+                error={!!error.password}
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: why !! ?

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                }}
+                variant="outlined"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                label="Email Address"
+                id="outlined-start-adornment"
+                value={email}
+                error={!!error.email}
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: why !!

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+                }}
+                variant="outlined"
+              />
+            </Grid>
+            <Grid item xs={6}>
+              <TextField
+                label="Confirm Password"
+                id="outlined-start-adornment"
+                type="password"
+                error={!!error.confirmPassword}
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: why '!!' ?

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+          >
+            Submit
+          </Button>
+        </DialogActions>
+      </Dialog>
+    );
+  }
+}
+
+export default withStyles(useStyles)(AddDialog);
```
Comment (Prabal555) [inline src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: Please make sure that export default is the last statement of the file

File: src/pages/Trainee/Trainee.jsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import Button from '@material-ui/core/Button';
+import AddDialog from './Components/AddDialog/index';
```
Comment (Prabal555) [inline src/pages/Trainee/Trainee.jsx]: the Components folder should be of lower case components

**Review by Prabal555 — CHANGES_REQUESTED** [review]

File: src/pages/Trainee/Components/AddDialog/AddDialog.jsx (hunk 4 above)
Comment (nxfareed) [reply src/pages/Trainee/Components/AddDialog/AddDialog.jsx]: To disable the submit button first and then to enable it after all the conditions are fulfilled.

**Pull request merged**